cabal-install/src/Distribution/Client/Utils/Parsec.hs

```diff
@@ -167,3 +170,23 @@ instance (Newtype a b, Ord a, Sep sep, Parsec b) => Parsec (NubList' sep b a) wh
 
 instance (Newtype a b, Sep sep, Pretty b) => Pretty (NubList' sep b a) where
   pretty = prettySep (Proxy :: Proxy sep) . map (pretty . (pack :: a -> b)) . NubList.fromNubList . unpack
+
+newtype NumJobs = NumJobs {getNumJobs :: Maybe Int}
+
+instance Newtype (Maybe Int) NumJobs
+
+instance Parsec NumJobs where
+  parsec = parsecNumJobs
+
+parsecNumJobs :: CabalParsing m => m NumJobs
+parsecNumJobs = ncpus <|> numJobs
```

Comment:
See `Distribution.Setup.Simple.Common.numJobsParser` for original, non-parsec parser